project/src/main/resources/ScapegoatInspections.scala

```diff
@@ -20,7 +20,7 @@ package scapegoat
 
 import org.sonar.api.batch.rule.Severity
 
-private[scapegoat] sealed trait Level {
+private[scapegoat] sealed trait Level extends Product with Serializable {
```

Comment:
https://typelevel.org/blog/2018/05/09/product-with-serializable.html

TL;DR; All case classes / objects  `extend Product with Serializable`, thus it is a good idea to make the root trait to extend it, that help a little bit to type inference, specially when using collections with **ADTs** .

On our case we do not need it, but now i am just used to do it when defining **ADTs**, is like a personal best practice. However, If you want, I can remove it since I do not have a strong argument for this case.

Reply:
Tbh, I've never seen before anyone else do it and I think the problem goes away when you explicitly give types to your values, which is what I tend to do and that's probably why I've never run into this issue. But let's keep it if that's your preference.